app/models/merchant.rb

```diff
@@ -20,4 +20,11 @@ def revenue
     .sum('invoice_items.quantity * invoice_items.unit_price')
   end
 
+  def self.total_revenue(date)
+    Invoice
```

Comment:
Instead of starting with the class Invoice, I'm wondering about starting with ```joins(invoices: [:invoice_items, :transactions]).......```, since there's a relationship set up between merchants and invoices. This feels more encapsulated since we would no longer be referencing the class Invoice inside of Merchant.

I'm not sure if it'll work (haven't poked around in the console yet with it), so it may be a thought to file away.